tests/utils/helpers.py

```diff
@@ -17,6 +20,45 @@
 )
 
 
+def rlp_encode_tx(tx: dict) -> bytes:
+    if "type" in tx:
+        typed_transaction = TypedTransaction.from_dict(tx)
+
+        sanitized_transaction = transaction_rpc_to_rlp_structure(
+            typed_transaction.transaction.dictionary
+        )
+
+        # RPC-structured transaction to rlp-structured transaction
+        rlp_serializer = (
+            typed_transaction.transaction.__class__._unsigned_transaction_serializer
+        )
+        encoded_unsigned_tx = pipe(
+            rlp_serializer.from_dict(sanitized_transaction),  # type: ignore  # noqa: E501
+            lambda val: rlp.encode(val),  # rlp([...])
+            lambda val: bytes(
+                [typed_transaction.transaction.__class__.transaction_type]
+            )
+            + val,  # (0x01 || rlp([...]))
+        )
+
```

Comment:
what's the diff with `transaction.payload()`? `serializable_unsigned_transaction_from_dict` from `eth_account/_utils/legacy_transactions.py` might also be just what you're looking for, handling all tx types

Reply:
the `payload` method includes the signature in it

```python
if not all(k in self.dictionary for k in "vrs"):
            raise ValueError("attempting to encode an unsigned transaction")
```


`serializable_unsigned_transaction_from_dict` isn't doing exactly what I want because in the case of a typed tx, it's simply returning an instance of the TypedTransaction object